_posts/2023-11-14-setup-microfrontend.md

```diff
@@ -0,0 +1,215 @@
+---
+title: "Angular & Micro Frontends"
+excerpt: "Micro Frontends is a modern architectural approach for developing web applications."
+tags: Micro Frontends
+authors:
+  - Harsh Julka
+header:
+  teaser: /assets/images/post/angular_microfrontends.png
+  teaser_alt: Micro Frontends
+category: Frontend
+---
+
+![](/assets/images/post/angular_microfrontends.png)
+
+# Introduction
+Micro Frontends is a modern architectural approach for developing web applications. Instead of creating a large, monolithic web application, it involves building multiple smaller frontend applications. Each of these small applications is responsible for a specific feature, which can be independently deployed and scaled. These Micro Frontends interact with each other to create a seamless user interface. This approach allows for greater flexibility and easier maintenance of web applications.
+
+Micro Frontends Architecture is composed of two or more applications
+- Shell/Host Application
+- Mutiple Remote applications.
+
+Transform your Angular application into a powerful micro frontend with ease. Simply add these essential dependencies to your project and start unlocking the full potential of micro frontends.
+- `@angular-architects/module-federation:` This provides support for webpack module federation plugin.
+- `@angular/elements:` Angular elements uses customElementRegistry interface which provides methods for registering custom elements and querying registered elements.
+
+```
+Note: Module federation support starts from webpack 5 and Angular 11
+```
+
+Without delay let's jump to a practical example.
+
+### Break monolithic app into Shell and Remote application.
+
+![](/assets/images/post/monolithic.png)
+
+**Convert our existing project to a shell application.**
+
+Use below command to add module federation dependency in shell application.
+```
+ng add @angular-architects/module-federation@<version> --project projectName --port shellPort --type host
+```
+
+`Use relevant version of @angular-architects/module-federation.` 
+<a href="https://www.npmjs.com/package/@angular-architects/module-federation#which-version-to-use" style="color:#1d42b3; text-decoration: none;" target="_blank">check here version compatibility.</a>
```

Comment:
Is this inline style intended? Having a different style for a link may cause accessibility issues